core/chains/evm/headtracker/head_listener.go

```diff
@@ -121,12 +162,14 @@ func (hl *headListener) receiveHeaders(ctx context.Context, handleNewHead httype
 			if !open {
 				return errors.New("head listener: chHeaders prematurely closed")
 			}
-			if blockHeader == nil {
+			if blockHeader.Equals(hl.getNilHead()) {
```

Comment:
Would flipping the check make sense here too? For example:
```suggestion
			if blockHeader.IsValid() {
```
'Valid' would be more flexible, for types that are not nilable, but may want the zero value or some custom logic to serve the same purpose.

Reply:
thanks for that note. I think having `IsValid()` function allows for more flexibility. Will update to that!

Reply:
Hmm i was thinking about this change, and the issue would be if blockHeader was `nil`, it would run into runtime error as `blockHeader` can't call `IsValid()` if its nil.

Reply:
Does that mean if `h *Head` was nil, methods can still be callable by h?

like so:
```
func (h *Head) IsValid() bool {
	return h != nil
}
```